pkg/bus/meseeks.go

```diff
@@ -86,6 +122,49 @@ func (b *Bus) seekRequest(ctx context.Context, auth, userID, connID string,
 	return nil
 }
 
+func (b *Bus) updateSeekRequest(ctx context.Context, auth, userID, connID string,
+	newReq *pb.SeekRequest) error {
+
+	// If we are here the seek exists and the game request is not nil
+	if newReq.GameRequest == nil {
+		return errors.New("nil game request for seek to update")
+	}
+	reqId := newReq.GameRequest.RequestId
+	sg, err := b.soughtGameStore.Get(ctx, reqId)
+	if err != nil {
+		return err
+	}
+
+	if userID == sg.Seeker() {
+		sg.SeekRequest.UserState = newReq.UserState
+		sg.SeekRequest.BootedReceivers = newReq.BootedReceivers
+	} else {
+		sg.SeekRequest.ReceiverState = newReq.ReceiverState
+	}
+
+	// If both players are ready, start the game
+	// If the receiver is absent, send a new seek to everyone
+	// Otherwise, only send it to the two players
+	if sg.SeekRequest.ReceiverState == pb.SeekState_READY &&
+		sg.SeekRequest.UserState == pb.SeekState_READY {
+		return b.gameAccepted(ctx, &realtime.SoughtGameProcessEvent{RequestId: reqId}, userID, connID)
+	} else if sg.SeekRequest.ReceiverState == pb.SeekState_ABSENT {
+		// Send to everyone, the seek is open
+		return b.newSeekRequest(ctx, auth, userID, connID, newReq)
+	} else {
+		// Update the current seek request
```

Comment:
Two cases for sending the seek:

- Send to the whole world (there is an ABSENT state or receiver goes from absent to not absent)
- Only send to players inwolwed (otherwise)